### DISCUSSION TITLE
File Explorer APIs / Events

### DISCUSSION THREAD

**Discussion by freddysilber** [microsoft/vscode-discussions#1115] [Extension Development QnA] [answered]
I was wanting to build an extension based on the file explorer in the side panel. 

My goal is that when I hover over a file in the explorer, then I could display a preview of the file of some kind in the main editor or a popover etc.

I'm having a tough time trying to find some API that I can use to interact with the explorer in this way.

If someone could point me in the right direction that would be great! I'm stumped.

**Comment by isidorn** [comment, marked as answer]
This API does not exist. Sorry!

You could use another API to register a handler for a specific file type. You can check how Markdown does the Markdown preview experience for example.
